heft-plugins/heft-api-extractor-plugin/src/ApiExtractorPlugin.ts

```diff
@@ -0,0 +1,246 @@
+        await this._getApiExtractorConfigurationAsync(taskSession, heftConfiguration);
+      if (apiExtractor && apiExtractorConfiguration) {
+        await this._runApiExtractorAsync(
+          taskSession,
+          heftConfiguration,
+          runOptions,
+          apiExtractor,
+          apiExtractorConfiguration
+        );
+      }
+    });
+  }
+
+  private async _getApiExtractorConfigurationFilePathAsync(
+    heftConfiguration: HeftConfiguration
+  ): Promise<string | undefined> {
+    if (!this._apiExtractorConfigurationFilePathPromise) {
+      this._apiExtractorConfigurationFilePathPromise =
+        heftConfiguration.rigConfig.tryResolveConfigFilePathAsync(CONFIG_FILE_LOCATION);
+    }
+    return await this._apiExtractorConfigurationFilePathPromise;
+  }
+
+  private async _getApiExtractorConfigurationAsync(
+    taskSession: IHeftTaskSession,
+    heftConfiguration: HeftConfiguration,
+    ignoreMissingEntryPoint?: boolean
+  ): Promise<TApiExtractor.ExtractorConfig | undefined> {
+    const configObjectFullPath: string | undefined = await this._getApiExtractorConfigurationFilePathAsync(
+      heftConfiguration
+    );
+    if (!configObjectFullPath) {
+      return undefined;
+    }
+
+    const apiExtractor: typeof TApiExtractor = (await this._getApiExtractorAsync(
+      taskSession,
+      heftConfiguration
+    ))!;
+    const configObject: TApiExtractor.IConfigFile =
+      apiExtractor.ExtractorConfig.loadFile(configObjectFullPath);
+
+    return apiExtractor.ExtractorConfig.prepare({
+      configObject,
+      configObjectFullPath,
+      ignoreMissingEntryPoint,
+      packageJsonFullPath: path.join(heftConfiguration.buildFolder, 'package.json'),
+      projectFolderLookupToken: heftConfiguration.buildFolder
+    });
+  }
+
+  private async _getApiExtractorAsync(
+    taskSession: IHeftTaskSession,
+    heftConfiguration: HeftConfiguration
+  ): Promise<typeof TApiExtractor | undefined> {
+    if (!this._apiExtractorPromise) {
+      this._apiExtractorPromise = this._getApiExtractorInnerAsync(taskSession, heftConfiguration);
+    }
+    return await this._apiExtractorPromise;
+  }
+
+  private async _getApiExtractorInnerAsync(
+    taskSession: IHeftTaskSession,
+    heftConfiguration: HeftConfiguration
+  ): Promise<typeof TApiExtractor | undefined> {
+    // API Extractor provides an ExtractorConfig.tryLoadForFolder() API that will probe for api-extractor.json
+    // including support for rig.json.  However, Heft does not load the @microsoft/api-extractor package at all
+    // unless it sees a config/api-extractor.json file.  Thus we need to do our own lookup here.
+    const apiExtractorConfigurationFilePath: string | undefined =
+      await this._getApiExtractorConfigurationFilePathAsync(heftConfiguration);
+    if (!apiExtractorConfigurationFilePath) {
+      return undefined;
+    }
+
+    const apiExtractorPackagePath: string = await heftConfiguration.rigToolResolver.resolvePackageAsync(
```

Comment:
`rigToolResolver` is a slightly confusing name.

Reply:
I've also named the class this... any other ideas? `RigPackageResolver` maybe? `RigResolver`?

Reply:
I like `RigPackageResolver`.